packages/react-charting/src/components/TreeChart/TreeChart.base.tsx

```diff
@@ -120,22 +122,30 @@ class StandardTree {
     // Text position x = x + rectWidth/2, 2 is ratio for length
     // Sub-text position x = x + rectWidth/2, 2 is ratio for length
 
-    const subValue =
-      metricName !== undefined ? (
-        <tspan className={this.styleClassNames.rectmetricText} dy="1.4em" x={xCoordinate + rectangleWidth / 2}>
-          {metricName}
-        </tspan>
-      ) : (
+    const subValue = metricName ? (
+      <tspan className={this.styleClassNames.rectMetricText} dy="1.4em" x={xCoordinate + rectangleWidth / 2}>
+        {metricName}
+      </tspan>
+    ) : bodytext ? (
+      <>
         <tspan className={this.styleClassNames.rectSubText} dy="1.4em" x={xCoordinate + rectangleWidth / 2}>
           {subname}
         </tspan>
-      );
+        <tspan className={this.styleClassNames.rectBodyText} dy="1.4em" x={xCoordinate + rectangleWidth / 2}>
+          {bodytext}
+        </tspan>
+      </>
+    ) : (
+      <tspan className={this.styleClassNames.rectSubText} dy="1.4em" x={xCoordinate + rectangleWidth / 2}>
+        {subname}
```

Comment:
Why is subname twice here.

Reply:
Here we are checking all composition , subname is twice because firstly we are checking for if optional bodyname is there then we are placing it below subname and for last case when it is not there so there will only be subname.